### PR TITLE
Docs: Moving the React "Using the HotColumn component" examples from CodeSandbox to the HoT docs

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR continues #8962 :
- Moves the React examples from the [Using the HotColumn component](https://handsontable.com/docs/react-hot-column/) page from CodeSandbox to the Handsontable docs

### Declaring column settings (done in #8962 )
- [Old CodeSandbox example](https://codesandbox.io/embed/declaring-column-settings-seg0l?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#declaring-column-settings)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)

### Declaring a custom renderer as a component (done in #8962 )
- [Old CodeSandbox example](https://codesandbox.io/embed/declaring-a-custom-renderer-as-a-component-4yv9m?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#declaring-a-custom-renderer-as-a-component)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)
  - I had to "merge" the separate `rendererComponent.jsx` file into the main code snippet

### Object data source
- [Old CodeSandbox example](https://codesandbox.io/embed/object-data-source-y8xwr?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#object-data-source)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)
  - I had to "merge" the separate component files into the main code snippet

### Declaring a custom editor as a component
- [Old CodeSandbox example](https://codesandbox.io/embed/declaring-a-custom-editor-as-a-component-qhbnv?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#declaring-a-custom-editor-as-a-component)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)

### Using the renderer/editor components within React's Context
- [Old CodeSandbox example](https://codesandbox.io/embed/using-the-renderer-component-with-reacts-context-forked-27pl0?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#using-the-renderer-editor-components-with-react-s-context)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)
  - After moving to the `Preview` tab, you can't click on the previous tabs anymore

### An advanced example
- [Old CodeSandbox example](https://codesandbox.io/embed/advanced-handsontablereact-implementation-using-hotcolumn-forked-jgrk2?fontsize=14&theme=dark)
- [New docs example](http://localhost:8080/docs/next/react-hot-column/#an-advanced-example)
- Seems to be working fine :white_check_mark:, but:
  - The `--tab preview` option doesn't seem to work (the `Preview` tab should display as default)

[skip changelog]

### CODE REVIEW

**Review by warpech — CHANGES_REQUESTED** [review]
I took it as a challenge to fix the remaining snippets. Please review if works for you now. The example 6 will not work in JSFiddle unless you deploy it on our server (because it needs the current changes in `fixes.js` to be deployed.

Below is one additional suggestion.

File: docs/11.0/guides/integrate-with-react/react-hot-column.md [hunk 1]
```diff
@@ -103,18 +102,127 @@ ReactDOM.render(<App />, document.getElementById('example2'));
...
+      name: "Kate",
+      score: 61,
+      isPromoted: true
+    },
+    {
+      id: 5,
+      name: "Max",
+      score: 98,
+      isPromoted: true
+    },
```
[the remark this answers is not in the document]
Comment (kirszenbaum) [reply docs/11.0/guides/integrate-with-react/react-hot-column.md:168]: @warpech - done: 43f0a8e (I also fixed the order of the IDs)

[the remark this answers is not in the document]
Comment (warpech) [reply docs/11.0/guides/integrate-with-react/react-hot-column.md:168]: Thanks!

**Comment by warpech** [comment]
I will not be able to help with the problems listed in the last comment (and probably not all are problems). 

@jansiegel can you help out?

**Review by warpech — DISMISSED** [review]
I discussed with @kirszenbaum that the changes can be accepted now. The remaining problems are tracked in new issues, and can be fixed later with high priority.

**Review by warpech — APPROVED** [review]

**Comment by jansiegel** [comment]
@kirszenbaum @warpech 
As for:
> In the 5th example, after clicking on the Preview tab, you're stuck: you can't click on the remaining tabs anymore

it seems to be caused by the styles of the `h1 - h3` headers. The `<h3>` header's margin rendered in this example covers the tab buttons, making them impossible to be clicked.

The problematic fragment is here: https://github.com/handsontable/handsontable/blob/develop/docs/.vuepress/theme/styles/index.styl#L139

I'd suggest changing
```
h1 {
```
to something like  this (for `h1`, `h2` and `h3`):
```
.theme-default-content > h1 {
```
if it makes sense in the current theme. 

Also, there's quite a lot of `!important`s in the styles - I'd try to get rid of those, as they'll most probably eventually cause some issues.

**Pull request merged**